packages/workflow/src/Interfaces.ts

```diff
@@ -1250,6 +1250,16 @@ export interface INodeType {
 	};
 }
 
+/**
+ * This class serves as the base for all nodes using the new context API
+ * having this as a class enables us to identify these instances at runtime
+ */
+export abstract class Node {
+	abstract description: INodeTypeDescription;
+	execute?(context: IExecuteFunctions): Promise<INodeExecutionData[][]>;
```

Comment:
Return type from INodeType is "Promise<INodeExecutionData[][] | NodeExecutionWithMetadata[][] | null>"

Reply:
intentionally changed this because it seemed like most of the nodes return an empty array. So, the ones that return `null` could be updated to return `[]` instead as well, hence simplifying the code a bit.

Reply:
Not sure if that would cause problems. Not 100% sure anymore but I think the difference between returning `null` and `[]` is that with an empty array the node does count as executed, while with `null` not. Was for example used by the old SplitInBatches node.

Reply:
@janober since this only affects the nodes that we port over to this pattern, do we still want to continue supporting returning `null`, and if so, should we make that behavior consistent across all nodes?